proto/errors.proto

```diff
@@ -26,7 +26,7 @@ import "gogoproto/gogo.proto";
 // A NotLeaderError indicates that the current range is not the
 // leader. If the leader is known, its Replica is set in the error.
 message NotLeaderError {
-  optional Replica leader = 1 [(gogoproto.nullable) = false];
+  optional Replica leader = 1;
```

Comment:
@tschottdorf I just merged with your changes and had to correct something in dist_sender.go which was expecting a non-nil leader value in the NotLeaderError error. Could you review that code and see if the possibility of a nil leader impacts the correctness?